smallfoot/app.py

```diff
@@ -0,0 +1,258 @@
+        #self.selection.selection_expr = None
+        #self.filter_ = self.filter_*0.0
+        time_cube=np.load('.//images_and_data//'+self.select_file.value)[:-1,:,:]
+        print('new shape:',time_cube.shape)
+        self.ds = utils.time_cube_to_xarray(time_cube)
+        
+        self.filter_ = np.zeros(
+            (max(time_cube.shape[:2]), max(time_cube.shape[:2]))
+        )
+        self.selection.selection_expr = None
+        #_filter = self.selection(hv.DynamicMap(self._spectrum_graph))
+        #filter_graphs = _filter + hv.DynamicMap(self._update_filter)
+        #filter_graphs = filter_graphs.opts(plot=dict(shared_axes=False))
+  
+        return self
+
+    def _update_self(self, event=None):
+        #print('old shape',self.ds.shape)
+        print('change file:',self.select_file.value,self.ds.isel(time_slice=self.time_slice_deep_slider.value).amplitude.values.shape,self.filter_.shape)
+        time_cube=np.load('.//images_and_data//'+self.select_file.value)[:-1,:,:]
+        #time_cube=np.load('.//images_and_data//'+'penobscot.npy')[:-1,:,:]
+        print('new shape:',time_cube.shape)
+        self.ds = utils.time_cube_to_xarray(time_cube)
+        self.time_slice_slider.end=len(self.ds.time_slice) - 1
+        self.time_slice_slider.value=0
+        self.time_slice_deep_slider.value =len(self.ds.time_slice) - 5 
+        self.time_slice_deep_slider.end = len(self.ds.time_slice) - 1
+        self.filter_ = np.zeros(
+            (max(time_cube.shape[:2]), max(time_cube.shape[:2]))
+        ) 
+        self.selection.selection_expr = None
+
+        return self
+
+    def app(self):
+        graphs = self.main_graphs()
+        options = self.options_pane()
+        template = pn.template.MaterialTemplate(
+            title="F.R.I.D.A. Seismic Footprint Filter Designer App",
+            logo="https://images.squarespace-cdn.com/content/58a4b31dbebafb6777c575b4/1549829488328-IZMTRHP7SLI9P9Z7MUSW/website_logo_head.png?content-type=image%2Fpng",
+            # logo="https://raw.githubusercontent.com/CEREGE-CL/CEREGE-CL.github.io/main/logo.png",
+            # favicon="https://raw.githubusercontent.com/CEREGE-CL/CEREGE-CL.github.io/main/logo.png",
+            header_background="#42a5f5",
+        )
+        template.sidebar.append(options)
+        template.main.append(graphs)
+        return template
+
+
+if "bokeh_app" in __name__:
+#    seismic = np.load("images_and_data/penobscot.npy")
+#     seismic = np.load("images_and_data/F3_original_subvolume_IL230-430_XL475-675_T200-1800.npy")
+    listfile = []
+    listname = []
+    for root, dirs, files in os.walk(r'./images_and_data/'):
+        for file in files:
+            file2=file.lower()
+            if file2.endswith(".sgy") or file2.endswith(".npy"):
+               listfile.append(os.path.join(root, file))
+               listname.append(file)
+    fd = FilterDesigner(listfile)
+    fd.app().servable("Filter Designer")
```

Comment:
I am going to have to merge and run the app to really be able to test.

Reply:
I think you can test it by cloning this branch first @mycarta :
`git clone -b panel_fix https://github.com/mycarta/t21-hack-footprint/`

Reply:
I will try it this weekend @leocd91